In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os

In [2]:
dataList = []
for filename in os.listdir ("Daily Stock Price  Returns104600698"):
    if filename[-3:] == "csv":
        head = r"Daily Stock Price  Returns104600698/"
        data = pd.read_csv(head + filename, index_col=0)
        dataList.append(data)
for filename in os.listdir ("Daily Stock Price  Returns165304594"):
    if filename[-3:] == "csv":
        head = r"Daily Stock Price  Returns165304594/"
        data = pd.read_csv(head + filename, index_col=0)
        dataList.append(data)
data = pd.concat(dataList, axis = 0)

Stkcd [Stock Code] - Codes released by the Shanghai and Shenzhen Stock Exchanges\
Trddt [Trading Date] - Denoted by YYYY-MM-DD\
Dretwd [Daily Return With Cash Dividend Reinvested] - Issue price is used as previous closing price of a stock on its first trading day. Please refer to “How Returns Are Calculated”.\

In [3]:
data

Trddt    Dretwd
Stkcd                       
1       2013-01-04 -0.001873
1       2013-01-07  0.019387
1       2013-01-08 -0.018405
1       2013-01-09 -0.008750
1       2013-01-10  0.000631
...            ...       ...
900957  2022-12-26  0.021779
900957  2022-12-27  0.001776
900957  2022-12-28 -0.007092
900957  2022-12-29 -0.005357
900957  2022-12-30  0.012567

[8210216 rows x 2 columns]

I use the index in eps to exclude the non-mainboard stocks.

In [4]:
eps_data = pd.read_csv("Processed_EPS.csv")

In [5]:
eps_data.Stkcd.unique

<bound method Series.unique of 0             1
1             1
2             1
3             1
4             1
          ...  
35182    605333
35183    605336
35184    605358
35185    605366
35186    605399
Name: Stkcd, Length: 35187, dtype: int64>

In [6]:
data = data.loc[eps_data.Stkcd.unique()]

In [7]:
data

Trddt    Dretwd
Stkcd                       
1       2013-01-04 -0.001873
1       2013-01-07  0.019387
1       2013-01-08 -0.018405
1       2013-01-09 -0.008750
1       2013-01-10  0.000631
...            ...       ...
605399  2022-12-26  0.055997
605399  2022-12-27  0.008135
605399  2022-12-28 -0.019725
605399  2022-12-29 -0.008232
605399  2022-12-30 -0.007070

[6022126 rows x 2 columns]

In [8]:
market_return = pd.read_csv("Aggregated Daily Market Returns204344531/TRD_Cndalym.csv", index_col=0)

In [9]:
market_return

Trddt  Cdretwdeq
Markettype                       
10          2013-01-04   0.014720
5           2013-01-04  -0.003810
15          2013-01-04  -0.002974
21          2013-01-04  -0.002965
31          2013-01-04  -0.002279
...                ...        ...
95          2022-12-30   0.007401
101         2022-12-30   0.006584
111         2022-12-30   0.006558
117         2022-12-30   0.006544
127         2022-12-30   0.006525

[46170 rows x 2 columns]

Markettype [Aggregated Market Type] - 5= SSE-SZSE A share market (excluding STAR Market, ChiNext); 10= SSE-SZSE B share market; 15= SSE-SZSE A & B share market; 21= SSE-SZSE A share market and ChiNext; 31= SSE-SZSE A & B share market and ChiNext; 37= SSE-SZSE A share market and STAR Market; 47= SSE-SZSE A & B share market and STAR Market; 53= SSE-SZSE A share market and ChiNext and STAR Market; 63= SSE-SZSE A & B share market and ChiNext and STAR Market; 33= SSE A share market and STAR Market; 20= SZSE A share market and ChiNext; 69= SSE-SZSE-BSE A share market; 79= SSE-SZSE-BSE A & B share market; 85= SSE-SZSE-BSE A share market and ChiNext; 95= SSE-SZSE-BSE A & B share market and ChiNext; 101= SSE-SZSE-BSE A share market and STAR Market; 111= SSE-SZSE-BSE A & B share market and STAR Market; 117= SSE-SZSE-BSE A share market and ChiNext and STAR Market; 127= SSE-SZSE-BSE A & B share market and ChiNext and STAR Market\
Trddt [Trading Date] - Denoted by YYYY-MM-DD\
Cdretwdos [Daily Aggregated Market Returns With Cash Dividend Reinvested (Current-Value-Weighted)] - Please refer to “How Returns Are Calculated”

In [10]:
market_return = market_return.loc[5]

In [11]:
market_return = market_return.set_index("Trddt")

In [12]:
data = data.reset_index()
data = data.set_index("Trddt")

In [13]:
data

Stkcd    Dretwd
Trddt                       
2013-01-04       1 -0.001873
2013-01-07       1  0.019387
2013-01-08       1 -0.018405
2013-01-09       1 -0.008750
2013-01-10       1  0.000631
...            ...       ...
2022-12-26  605399  0.055997
2022-12-27  605399  0.008135
2022-12-28  605399 -0.019725
2022-12-29  605399 -0.008232
2022-12-30  605399 -0.007070

[6022126 rows x 2 columns]

In [14]:
 def merge_market(data: pd.DataFrame) -> pd.DataFrame:
    print(data.index[0])
    data.loc[:, "AR"] = data.loc[:, "Dretwd"] - market_return.loc[data.index[0]][0]
    data.to_csv(r"individual_stock_return/" + data.index[0])

In [16]:
new_data = data.groupby("Trddt").apply(merge_market)

2013-01-04
2013-01-07
2013-01-08
2013-01-09
2013-01-10
2013-01-11
2013-01-14
2013-01-15
2013-01-16
2013-01-17
2013-01-18
2013-01-21
2013-01-22
2013-01-23
2013-01-24
2013-01-25
2013-01-28
2013-01-29
2013-01-30
2013-01-31
2013-02-01
2013-02-04
2013-02-05
2013-02-06
2013-02-07
2013-02-08
2013-02-18
2013-02-19
2013-02-20
2013-02-21
2013-02-22
2013-02-25
2013-02-26
2013-02-27
2013-02-28
2013-03-01
2013-03-04
2013-03-05
2013-03-06
2013-03-07
2013-03-08
2013-03-11
2013-03-12
2013-03-13
2013-03-14
2013-03-15
2013-03-18
2013-03-19
2013-03-20
2013-03-21
2013-03-22
2013-03-25
2013-03-26
2013-03-27
2013-03-28
2013-03-29
2013-04-01
2013-04-02
2013-04-03
2013-04-08
2013-04-09
2013-04-10
2013-04-11
2013-04-12
2013-04-15
2013-04-16
2013-04-17
2013-04-18
2013-04-19
2013-04-22
2013-04-23
2013-04-24
2013-04-25
2013-04-26
2013-05-02
2013-05-03
2013-05-06
2013-05-07
2013-05-08
2013-05-09
2013-05-10
2013-05-13
2013-05-14
2013-05-15
2013-05-16
2013-05-17
2013-05-20
2013-05-21
2013-05-22
2013-05-23
2013-05-24

2016-02-04
2016-02-05
2016-02-15
2016-02-16
2016-02-17
2016-02-18
2016-02-19
2016-02-22
2016-02-23
2016-02-24
2016-02-25
2016-02-26
2016-02-29
2016-03-01
2016-03-02
2016-03-03
2016-03-04
2016-03-07
2016-03-08
2016-03-09
2016-03-10
2016-03-11
2016-03-14
2016-03-15
2016-03-16
2016-03-17
2016-03-18
2016-03-21
2016-03-22
2016-03-23
2016-03-24
2016-03-25
2016-03-28
2016-03-29
2016-03-30
2016-03-31
2016-04-01
2016-04-05
2016-04-06
2016-04-07
2016-04-08
2016-04-11
2016-04-12
2016-04-13
2016-04-14
2016-04-15
2016-04-18
2016-04-19
2016-04-20
2016-04-21
2016-04-22
2016-04-25
2016-04-26
2016-04-27
2016-04-28
2016-04-29
2016-05-03
2016-05-04
2016-05-05
2016-05-06
2016-05-09
2016-05-10
2016-05-11
2016-05-12
2016-05-13
2016-05-16
2016-05-17
2016-05-18
2016-05-19
2016-05-20
2016-05-23
2016-05-24
2016-05-25
2016-05-26
2016-05-27
2016-05-30
2016-05-31
2016-06-01
2016-06-02
2016-06-03
2016-06-06
2016-06-07
2016-06-08
2016-06-13
2016-06-14
2016-06-15
2016-06-16
2016-06-17
2016-06-20
2016-06-21
2016-06-22

2019-04-22
2019-04-23
2019-04-24
2019-04-25
2019-04-26
2019-04-29
2019-04-30
2019-05-06
2019-05-07
2019-05-08
2019-05-09
2019-05-10
2019-05-13
2019-05-14
2019-05-15
2019-05-16
2019-05-17
2019-05-20
2019-05-21
2019-05-22
2019-05-23
2019-05-24
2019-05-27
2019-05-28
2019-05-29
2019-05-30
2019-05-31
2019-06-03
2019-06-04
2019-06-05
2019-06-06
2019-06-10
2019-06-11
2019-06-12
2019-06-13
2019-06-14
2019-06-17
2019-06-18
2019-06-19
2019-06-20
2019-06-21
2019-06-24
2019-06-25
2019-06-26
2019-06-27
2019-06-28
2019-07-01
2019-07-02
2019-07-03
2019-07-04
2019-07-05
2019-07-08
2019-07-09
2019-07-10
2019-07-11
2019-07-12
2019-07-15
2019-07-16
2019-07-17
2019-07-18
2019-07-19
2019-07-22
2019-07-23
2019-07-24
2019-07-25
2019-07-26
2019-07-29
2019-07-30
2019-07-31
2019-08-01
2019-08-02
2019-08-05
2019-08-06
2019-08-07
2019-08-08
2019-08-09
2019-08-12
2019-08-13
2019-08-14
2019-08-15
2019-08-16
2019-08-19
2019-08-20
2019-08-21
2019-08-22
2019-08-23
2019-08-26
2019-08-27
2019-08-28
2019-08-29
2019-08-30

2022-05-26
2022-05-27
2022-05-30
2022-05-31
2022-06-01
2022-06-02
2022-06-06
2022-06-07
2022-06-08
2022-06-09
2022-06-10
2022-06-13
2022-06-14
2022-06-15
2022-06-16
2022-06-17
2022-06-20
2022-06-21
2022-06-22
2022-06-23
2022-06-24
2022-06-27
2022-06-28
2022-06-29
2022-06-30
2022-07-01
2022-07-04
2022-07-05
2022-07-06
2022-07-07
2022-07-08
2022-07-11
2022-07-12
2022-07-13
2022-07-14
2022-07-15
2022-07-18
2022-07-19
2022-07-20
2022-07-21
2022-07-22
2022-07-25
2022-07-26
2022-07-27
2022-07-28
2022-07-29
2022-08-01
2022-08-02
2022-08-03
2022-08-04
2022-08-05
2022-08-08
2022-08-09
2022-08-10
2022-08-11
2022-08-12
2022-08-15
2022-08-16
2022-08-17
2022-08-18
2022-08-19
2022-08-22
2022-08-23
2022-08-24
2022-08-25
2022-08-26
2022-08-29
2022-08-30
2022-08-31
2022-09-01
2022-09-02
2022-09-05
2022-09-06
2022-09-07
2022-09-08
2022-09-09
2022-09-13
2022-09-14
2022-09-15
2022-09-16
2022-09-19
2022-09-20
2022-09-21
2022-09-22
2022-09-23
2022-09-26
2022-09-27
2022-09-28
2022-09-29
2022-09-30
2022-10-10